backend/src/org/commcare/util/FormDataUtil.java

```diff
@@ -48,12 +48,16 @@ public static String getTitleFromSession(UserSandbox userSandbox,
                 // the 1st computed datum... another good heuristic would be
                 // session.getPoppedStep().getId().startsWith("case_id")
                 datumValue = session.getPoppedStep().getValue();
-            } else if (datum instanceof EntityDatum && ((EntityDatum)datum).getLongDetail() != null) {
+            } else if (datum instanceof EntityDatum) {
                 String tmpDatumValue = session.getPoppedStep().getValue();
                 if (tmpDatumValue != null) {
                     datumValue = tmpDatumValue;
                 }
-                return new Pair<>((EntityDatum)datum, datumValue);
+                if (((EntityDatum)datum).getLongDetail() == null) {
+                    return new Pair<>(null, datumValue);
```

Comment:
When there isn't a long detail we don't want to fire https://github.com/dimagi/commcare-core/pull/382/files/6d0f6bf1ae7ee116b0d3d2cf00fb24ece89a834e#diff-ff89d6b08ec41ebd600e0eefe0d93ebfR36 but rather https://github.com/dimagi/commcare-core/pull/382/files/6d0f6bf1ae7ee116b0d3d2cf00fb24ece89a834e#diff-ff89d6b08ec41ebd600e0eefe0d93ebfR33 to get the name associated with a case